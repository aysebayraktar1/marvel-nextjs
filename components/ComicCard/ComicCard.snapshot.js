import { render } from '@testing-library/react';
import ComicCard from '../ComicCard/index';

it('renders homepage unchanged', () => {
  const { container } = render(<ComicCard />);
  expect(container).toMatchSnapshot();
});
