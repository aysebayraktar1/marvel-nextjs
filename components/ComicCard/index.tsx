import React, { FC } from "react";

const ComicCard: FC<IComicCard> = ({ id, title, description }) => {
  return (
    <li key={id} data-testid="ComicCard__Component">
      {title}
      <pre></pre>
      <div dangerouslySetInnerHTML={{ __html: description }} />
    </li>
  );
};

export default ComicCard;