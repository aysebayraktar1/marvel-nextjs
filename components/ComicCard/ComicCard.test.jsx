/**
 * @jest-environment jsdom
 */

 import React from "react";

 import { render, cleanup } from "@testing-library/react";
 
 import ComicCard from "./index";
 
 let getByTestId;
 let component;
 
 beforeEach(() => {
   component = render(<ComicCard />);
   getByTestId = component.getByTestId;
 });
 
 afterEach(() => {
   getByTestId = null;
   component = null;
   cleanup();
 });
 
 describe("ComicCard component", () => {
   it("should rendering on screen", () => {
     const actual = getByTestId("ComicCard__Component");
     expect(actual).toBeInTheDocument();
   });
 });
 