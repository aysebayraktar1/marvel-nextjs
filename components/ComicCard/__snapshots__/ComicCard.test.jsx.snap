// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders comic card unchanged 1`] = `
<div>
  <li
    data-testid="ComicCard__Component"
  >
    <pre />
    <div />
  </li>
</div>
`;
